frontend/src/components/createProjectForm.tsx

```diff
@@ -0,0 +1,282 @@
+
+    let skillTypes: ISkillType[] = skillTypeResponse.data || [];
+    let skillTypeError: Error = skillTypeResponse.error;
+
+    const { t } = useTranslation("common");
+
+    const [coaches, setCoaches] = useState<string[] | []>([]);
+    const [hasPickedCoach, setHasPickedCoach] = useState<boolean>(false);
+    const [selectedCoach, setSelectedCoach] = useState<string>("");
+    const [skills, setSkills] = useState<string[] | []>([]);
+    const [selectedSkill, setSelectedSkill] = useState<string>("");
+    const [skillInfos, setSkillInfos] = useState<string[] | []>([]);
+    const [skillInfo, setSkillInfo] = useState<string>("");
+
+    if (userError) {
+        console.log(userError);
+        return null;
+    }
+
+    if (skillTypeError) {
+        console.log(skillTypeError);
+        return null;
+    }
+
+    let skillColorMap = new Map<string, string>();
+
+    for (let skillType of skillTypes) {
+        skillColorMap.set(skillType.name, skillType.colour);
+    }
+
+    function handleChangeSkill(e: ChangeEvent<HTMLInputElement>) {
+        setSelectedSkill(e.target.value);
+    }
+
+    function handleChangeSkillInfo(e: ChangeEvent<HTMLInputElement>) {
+        setSkillInfo(e.target.value);
+    }
+
+    function handleDeleteSkill(index: number) {
+        delete skills[index];
+        delete skillInfos[index];
+        setSkills(skills.filter((value) => value !== undefined));
+        setSkillInfos(skillInfos.filter((value) => value !== undefined));
+    }
+
+    function handleAddSkill() {
+        const newSkill = selectedSkill === "" ? skillTypes[0].name : selectedSkill;
+        const newSkills: string[] = (skills as string[]).concat(newSkill);
+
+        let newSkillInfo = skillInfo === "" ? capitalize(t("empty skill info")) : skillInfo;
+        const newSkillInfos = (skillInfos as string[]).concat(newSkillInfo);
+
+        setSkills(newSkills);
+        setSkillInfos(newSkillInfos);
+        setSkillInfo("");
+    }
+
+    function handleAddCoach(e: ChangeEvent<HTMLInputElement>) {
+        setHasPickedCoach(true);
+
+        if (!(coaches as string[]).includes(e.target.value)) {
+            const newCoaches: string[] = (coaches as string[]).concat(e.target.value);
+            setCoaches(newCoaches);
+        }
+
+        setSelectedCoach(e.target.value);
+    }
+
+    function handleDeleteCoach(index: number) {
+        delete coaches[index];
+        setCoaches(coaches.filter((value) => value !== undefined));
+    }
+
+    async function handleSubmit(values: ProjectCreationValues) {
+        values.skills = skills;
+        values.skillInfos = skillInfos;
+
+        const coachURLs: string[] = [];
+        for (let coach of coaches) {
+            coachURLs.push(
+                // @ts-ignore
+                // We know there will always be a user with this callname
+                users.find((item) => item.callName === coach)._links.self.href
```

Comment:
Not sure about the syntax but look up the use of `!` in TS.
```suggestion
                // We know there will always be a user with this callname
                users.find((item) => item.callName === coach)!._links.self.href
```